### DISCUSSION TITLE
How to save SVG files correctly without objects losing there position or losing data?

### DISCUSSION THREAD

**Discussion by john-nxt1sports** [fabricjs/fabric.js#9953] [Q&A]
Hi everyone, this is my first inquiry in here. I am new to fabric.js and have a issue.

When i upload SVG files into my application, they work, but sometimes the text is in wrong position, or there is content that will be missing. 

But more so, once I Save this SVG into my app. And return back to the template i just saved into my editor. The SVG file is all over the place, content will be missing, and it will be in the wrong positions.

How can i fix this two things:

1. When upload SVG, load in correct positions and without missing data?
2. When I save my SVG, to save it in correct position without losing data?

Note, the templates are typically pretty complex, as its for a sports graphic application that are made in XD or illustrator, them imported into our app for users to edit.
We have also tried optimizing SVG templates in CLI like SVGO and optimizing the SVG before upload into app without success, the SVGs have a lot of base64 data. ** We have had much more success with upload versus save, upload typically gets all the content to atleast get on the app then we reposition it.

Let me know, thank you very much. 🙏

**Comment by asturur** [comment]
The SVG is not meant for save and reload. Some SVG features are not implemented or can't be implemented.
For example a common one, TSPAN doesn't work.
To do save and reload operations use JSON export.
